test/stake/main.go

```diff
@@ -0,0 +1,325 @@
+package main
```

Comment:
Could you fit this into the single test/chain/main.go? These kind of code actually shouldn't exist in master repo, it's more like experimental code. So let's minimize the footprint of them.

Reply:
Yes, I agree. I have it as my todo.

Reply:
Done.